servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/UseDefaultLoadBalancerProvider.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.LoadBalancedConnection;
+import io.servicetalk.client.api.LoadBalancerFactory;
+import io.servicetalk.concurrent.api.Executor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Properties;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.loadbalancer.HealthCheckConfig.DEFAULT_HEALTH_CHECK_FAILED_CONNECTIONS_THRESHOLD;
+import static io.servicetalk.loadbalancer.HealthCheckConfig.DEFAULT_HEALTH_CHECK_INTERVAL;
+import static io.servicetalk.loadbalancer.HealthCheckConfig.DEFAULT_HEALTH_CHECK_JITTER;
+import static io.servicetalk.loadbalancer.HealthCheckConfig.DEFAULT_HEALTH_CHECK_RESUBSCRIBE_INTERVAL;
+import static io.servicetalk.loadbalancer.HealthCheckConfig.validateHealthCheckIntervals;
+import static io.servicetalk.utils.internal.NumberUtils.ensureNonNegative;
+
+public final class UseDefaultLoadBalancerProvider implements RoundRobinLoadBalancerBuilderProvider {
```

Comment:
My original expectation was that we will have only one provider - `DefaultHttpLoadBalancerProvider`, that will have a property to swap RRLB with DefaultLB with default values identical to RRLB behavior and properties to control more DefaultLB settings to try other features. Why do we need a 2nd provider?

Reply:
I suppose we never talked explicitly about it but this is how I envisioned implementing of the 'dual tracks' that we talked about: one track where we quietly and relatively quickly transition people over from RR to DefaultLB in RR compatibility mode and the second where users can explicitly try out new features that weren't available in RR. I kind of like your idea of a single flag but we should game it out a bit and I worry it could introduce some ambiguity between the dual tracks. In my mind these are the most important differentiators but I'm sure there are others worth considering.

### Pro's of single provider/flag
- Only a single flag is a fair degree simpler for users to think about. However, ideally they only need to know about the 'opt-in' flag and the one in this PR is for mostly just for us.
- We (or anybody else) don't need to consider interaction between two 'enable' flags.
- Less code and less complex story for partners that must know about both pathways.
### Pro's of two providers/flags
- Clear separation of what the extra fine-grained flags are associated with. Specifically, say we're mid way through migrating RR cases and have a customer that want's to use the flags to try P2C, but only for a single client. I don't think they could do that with only one 'enable' flag since changing the selection algorithm will change it for all clients.
- We don't have to worry about two different interests setting the same flag: ourselves trying to do a broad rollout for RR in compatibility mode and someone trying to test specific features.
- With a second opt-in flag users can test out the new load balancer with the default settings we want for the new load balancer: otherwise they'll need to adjust a bunch of settings to get to what we would like as the default. This one should be doable with some fancy property value parsing but it might get hairy fast.

The difficulty of running the parallel tracks is the most pressing problem to me. I suppose it's possible that if someone wants to experiment then it has the effect of disabling the RR -> Default LB in RR mode transition for non-selected clients although I'd prefer not to have back-sliding.

Reply:
I'm not opposed of having multiple flags or making them more fine-grade configurable (for example, per address). It's expected what there will be a lot more options bcz there are a lot more features on `DefaultLB`. My questions is only about implementation of these flags. If looks like if we already have one provider to configure `DefaultLB` instead of RRLB, we can reuse that and just add more props to drive different scenarios. Doesn't look like there is a need to use `RoundRobinLoadBalancerBuilderProvider`.

Reply:
Sorry for the misunderstanding. 🙂 

We discussed this offline. For posterity, we came to the conclusion that we need to use the `RoundRobinLoadBalancerBuilderProvider` if we want to honor any existing RR settings. We could fold the 'opt-in' provider to be a part of this one if we wanted so long as we can be sure to get the right `String` for `targetResource` but that will then start to include port info which we may not want. For now maybe merging this is a the path of least resistance.

@idelpivnitskiy, please correct me if I misinterpreted that part of our conversation, and thank you again for the review.